src/main/java/br/com/harbitech/school/config/security/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package br.com.harbitech.school.config.security;
+
+import br.com.harbitech.school.user.User;
+import br.com.harbitech.school.user.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+    private final UserRepository userRepository;
+
+    public UserDetailsServiceImpl(UserRepository userRepository){
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        Optional<User> user = userRepository.findByEmail(username);
+        if(user.isPresent()){
+            return user.get();
+        }
+        throw new UsernameNotFoundException("Dados inválidos");
```

Comment:
Isso aqui pode ser trocado por:

```suggestion
        return userRepository.findByEmail(username).orElseThrow(() -> new UsernameNotFoundException("Dados inválidos"));
```

Reply:
Realmente, bem melhor! Obrigada